### DISCUSSION TITLE
RaspAP Known WiFi Client Auto-connect

### DISCUSSION THREAD

**Discussion by Shilyjunior** [RaspAP/raspap-webgui#1563] [Q&A]
I'm using RaspAP with two WiFi networks: wlan1 for the AP and wlan0 as the client connecting to a network with access to the Internet and this works great. However, I will like to know if there is a way to set RasAP to auto connect to known wifi client on boothup in stead of having to do this manually every time.
<img width="614" alt="Screenshot 2024-04-07 at 09 54 16" src="https://github.com/RaspAP/raspap-webgui/assets/165553275/e138f53f-42e0-4a6b-8d23-ad1898660285">

**Comment by billz** [comment]
This is the default behavior of wpa_supplicant. That is, on startup the wpa_supplicant.service is executed by systemd (not RaspAP) and enables logging and the DBus control interface, nothing more.

`/lib/systemd/system/wpa_supplicant.service`
```
[Service]
Type=dbus
BusName=fi.w1.wpa_supplicant1
ExecStart=/sbin/wpa_supplicant -u -s -O "DIR=/run/wpa_supplicant GROUP=netdev"
```

The DBus interface is used to communicate with wpa_supplicant. You can have it auto connect on startup by adding the following to the root user's crontab:

```
$ sudo crontab -e

# m h  dom mon dow   command
@reboot /sbin/wpa_supplicant -B -Dnl80211 -c/etc/wpa_supplicant/wpa_supplicant.conf -iwlan0
```

**Reply by AlquistArj** [reply]
I realise setting a default can be done with wpa_supplicant but it would be a nice feature to have the ability to set a default, or to reconnect to the last used Wi-Fi client AP, from the web UI.

**Comment by Fontheweg** [comment]
I used this way to autoconnect ... but still got one problem.
After a few hours the connection is closed. I don't know why, but I can't reconnect. I have to reboot the RPi.

If there is a way the connection could be reconnect automatically too, it would be perfect.

I realized the autoconnect with a crontab that runs every minute and checks the interface connection. If it is not connected it makes a scan to lookup for known networks. If a known network is found it connects. But is the connection lost - no more scans are possible :(

**Reply by bananajoe86** [reply]
Would you share your crontab with us? I´m interested in it.

**Reply by Fontheweg** [reply]
It is a simple crontab that calls an python script every minute.

Actually I cant post the script because my SD Card is burned 🫣 and I didn't backup the stuff ... 😂
Testing ... damn!
If I have rewrote the script I can share it.

The simple version was:

`rm /var/run/wpa_supplicant/wlan1`
`sudo /sbin/wpa_supplicant -B -D nl80211 -c /etc/wpa_supplicant/wpa_supplicant.conf -i wlan1`

**Comment by Rohan726** [comment]
See if this helps: https://github.com/RaspAP/raspap-webgui/discussions/1817

P.S. Didnt need the "crontab -e" entry
